build.gradle

```diff
@@ -207,6 +207,17 @@ def distContent = copySpec {
   }
 }
 
+task buildDTrace(type:Exec) {
+  group 'Build'
+  enabled = (osName == 'SunOS')
+
+  workingDir 'make'
+
+  commandLine 'make'
+
+  dependsOn 'bootJar'
+}
+
```

Comment:
A script-layout thing: I would before the `distContent` copySpec. This way it would not disrupt the strong relationship between `distContent` and the two distribution tasks that use it.  
Using `make` this way could work (can't verify it on my box), but has a drawback: We are not taking advantage of gradle's ability to do incremental builds.  

To support incremental builds, you could declare the tasks inputs & outputs like this:
```groovy
task buildDTrace(type:Exec) {
  group 'Build'
  onlyIf {osName == 'SunOS'}
  inputs.dir 'src/solaris/native'
  inputs.files bootJar.outputs
  outputs.dir "${buildDir}/i386"

  workingDir 'make'

  commandLine 'make'
}
```